### PR TITLE
uc-crux-llvm: More flexible interface for registering overrides

### PR DESCRIPTION
**Pull request by langston-barrett**
This will eventually be exposed through `runSimulator`.

### CODE REVIEW

File: uc-crux-llvm/src/UCCrux/LLVM/Run/Simulate.hs [hunk 1]
```diff
@@ -83,14 +85,25 @@ import           UCCrux.LLVM.Errors.Panic (panic)
...
 {- ORMOLU_ENABLE -}
 
+newtype CreateOverrideFn arch =
+  CreateOverrideFn
+    { runCreateOverrideFn ::
+        forall sym.
+        IsSymInterface sym =>
+        HasLLVMAnn sym =>
+        sym ->
+        IO (PolymorphicLLVMOverride (Crux.Crux sym) sym arch)
```
Comment (travitch) [inline uc-crux-llvm/src/UCCrux/LLVM/Run/Simulate.hs:104]: Is this in IO for anticipated use cases? It wasn't obviously necessary below

File: uc-crux-llvm/src/UCCrux/LLVM/Overrides/Polymorphic.hs [hunk 2]
```diff
@@ -0,0 +1,75 @@
+{-
+Module       : UCCrux.LLVM.Overrides.Polymorphic
+Description  : A 'PolymorphicLLVMOverride' is one that's maximally polymorphic
+Copyright    : (c) Galois, Inc 2021
+License      : BSD3
+Maintainer   : Langston Barrett <langston@galois.com>
+Stability    : provisional
+-}
+
+{-# LANGUAGE GADTs #-}
```
Comment (travitch) [inline uc-crux-llvm/src/UCCrux/LLVM/Overrides/Polymorphic.hs:10]: Does this actually require GADTs?

Comment (langston-barrett) [reply uc-crux-llvm/src/UCCrux/LLVM/Overrides/Polymorphic.hs:10]: Yeah, `ArchOk` is an equational constraint.

File: uc-crux-llvm/src/UCCrux/LLVM/Overrides/Polymorphic.hs [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+{- ORMOLU_DISABLE -}
+import           Lang.Crucible.Backend (IsSymInterface)
+
+import           Lang.Crucible.LLVM.Intrinsics (OverrideTemplate)
+import           Lang.Crucible.LLVM.MemModel (HasLLVMAnn)
+
+import           Crux.LLVM.Overrides (ArchOk)
+{- ORMOLU_ENABLE -}
+
+-- | An override that is compatible with any suitable symbolic backend.
```
[the remark this answers is not in the document]
Comment (langston-barrett) [reply uc-crux-llvm/src/UCCrux/LLVM/Overrides/Polymorphic.hs:33]: Oh whoops yeah, that's a mistake! Thanks.

**Pull request merged**